commit a6af1d848179c17deb94621c2e761769f0d99355
Author: Paul Stewart <pstew@chromium.org>
Date:   Fri Jun 10 07:00:19 2011 -0800

    mac80211: Start monitor work on restart
    
    Trigger connection monitor on resume from suspend.  Since we
    have been sleeping, there is reason to suspect that we might
    not still be associated.  The speed of detecting loss of
    {connection,authentication} is worth the cost of the small
    additional traffic at resume.
    
    Signed-off-by: Paul Stewart <pstew@chromium.org>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index 0c6e9ef8c7f8..faca5033f061 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -2206,6 +2206,7 @@ void ieee80211_sta_restart(struct ieee80211_sub_if_data *sdata)
 		add_timer(&ifmgd->chswitch_timer);
 	ieee80211_sta_reset_beacon_monitor(sdata);
 	ieee80211_restart_sta_timer(sdata);
+	ieee80211_queue_work(&sdata->local->hw, &sdata->u.mgd.monitor_work);
 }
 #endif